src/ide/ide.ts

```diff
@@ -622,6 +622,8 @@ export let ide = (function () {
 				else module.programstate.stepping();
 			} else if (module.interpreter.status === "waiting")
 				module.programstate.waiting();
+			else if (module.interpreter.status === "dialog")
+				module.programstate.dialog();
```

Comment:
```suggestion
				module.programstate.waiting();
```

Why not just show "program is waiting" here? I don't think the dialog state is too special that we should expose it to the user (it's almost hidden behind the backdrop anyway). In addition, I'd find it more consistent if we indicated any kind of waiting as "program is waiting".